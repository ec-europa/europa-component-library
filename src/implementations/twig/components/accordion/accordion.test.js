import {
  merge,
  renderTwigFileAsNode,
  renderTwigFileAsHtml,
} from '@ecl/test-utils';
import { axe } from 'vitest-axe';

import demoData from '@ecl/specs-component-accordion/demo/data';

const oldData = JSON.parse(JSON.stringify(demoData));
oldData.icon.splice(1, 1);

describe('Accordion', () => {
  const template = '@ecl/accordion/accordion.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  test('renders correctly', () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test('renders correctly with old data', () => {
    expect.assertions(1);

    return expect(render(oldData)).resolves.toMatchSnapshot();
  });

  test('renders correctly with a single icon', () => {
    expect.assertions(1);

    oldData.icon[0].name = 'corner-arrow';

    return expect(render(oldData)).resolves.toMatchSnapshot();
  });

  test('renders correctly with extra class names', () => {
    expect.assertions(1);

    const optionsWithExtraClasses = merge(demoData, {
      extra_classes: 'custom-class custom-class--test',
    });

    return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
  });

  test('renders correctly with extra attributes', () => {
    expect.assertions(1);

    const optionsWithExtraAttrs = merge(demoData, {
      extra_attributes: [
        { name: 'data-test', value: 'data-test-value' },
        { name: 'data-test-1', value: 'data-test-value-1' },
      ],
    });

    return expect(render(optionsWithExtraAttrs)).resolves.toMatchSnapshot();
  });

  test(`passes the accessibility tests`, async () => {
    expect(
      await axe(await renderTwigFileAsHtml(template, demoData, true)),
    ).toHaveNoViolations();
  });
});
