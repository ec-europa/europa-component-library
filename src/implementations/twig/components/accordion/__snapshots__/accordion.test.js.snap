// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion renders correctly 1`] = `
<jest>
  <div
    class="ecl-accordion"
    data-ecl-accordion=""
    data-ecl-auto-init="Accordion"
  >
    <div
      class="ecl-accordion__item"
    >
      <h3
        class="ecl-accordion__title"
        id="accordion-example-title"
      >
        <button
          aria-controls="accordion-example-content"
          class="ecl-accordion__toggle"
          data-ecl-accordion-toggle=""
          type="button"
        >
          <span
            class="ecl-accordion__toggle-flex"
          >
            <span
              class="ecl-accordion__toggle-title"
            >
              Delivery of last pending proposals, a common Destiny of unity, the hour of European Democracy
            </span>
            <span
              class="ecl-accordion__toggle-indicator"
            >
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-accordion__toggle-icon"
                data-ecl-accordion-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#plus"
                />
              </svg>
            </span>
          </span>
        </button>
      </h3>
      <div
        aria-labelledby="accordion-example-title"
        class="ecl-accordion__content"
        hidden=""
        id="accordion-example-content"
        role="region"
      >
        The College of Commissioners held today the first weekly meeting of 2019 which was devoted to discussing the challenges of this new year. Commissioners used the opportunity to take stock and discuss the year ahead, including the European elections in May and other important milestones ahead.
      </div>
    </div>
    <div
      class="ecl-accordion__item"
    >
      <h3
        class="ecl-accordion__title"
        id="accordion-example2-title"
      >
        <button
          aria-controls="accordion-example2-content"
          class="ecl-accordion__toggle"
          data-ecl-accordion-toggle=""
          type="button"
        >
          <span
            class="ecl-accordion__toggle-flex"
          >
            <span
              class="ecl-accordion__toggle-title"
            >
              Spring 2019 Economic Forecast: Growth continues at a more moderate pace
            </span>
            <span
              class="ecl-accordion__toggle-indicator"
            >
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-accordion__toggle-icon"
                data-ecl-accordion-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#plus"
                />
              </svg>
            </span>
          </span>
        </button>
      </h3>
      <div
        aria-labelledby="accordion-example2-title"
        class="ecl-accordion__content"
        hidden=""
        id="accordion-example2-content"
        role="region"
      >
        The European economy is forecast to continue expanding for the seventh year in a row in 2019, with real GDP expected to grow in all EU Member States. As global uncertainties continue to weigh, domestic dynamics are set to support the European economy. Growth is expected to gather pace again next year.
      </div>
    </div>
    <div
      class="ecl-accordion__item"
    >
      <h3
        class="ecl-accordion__title"
        id="accordion-example3-title"
      >
        <button
          aria-controls="accordion-example3-content"
          class="ecl-accordion__toggle"
          data-ecl-accordion-toggle=""
          type="button"
        >
          <span
            class="ecl-accordion__toggle-flex"
          >
            <span
              class="ecl-accordion__toggle-title"
            >
              Delivery of last pending proposals, a common Destiny of unity, the hour of European Democracy
            </span>
            <span
              class="ecl-accordion__toggle-indicator"
            >
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-accordion__toggle-icon"
                data-ecl-accordion-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#plus"
                />
              </svg>
            </span>
          </span>
        </button>
      </h3>
      <div
        aria-labelledby="accordion-example3-title"
        class="ecl-accordion__content"
        hidden=""
        id="accordion-example3-content"
        role="region"
      >
        In the modern global economy trade is essential for growth, jobs and competiveness, and the EU is committed to maintaining an open and rules-based trading system. With the rising threat of protectionism and weakened commitment of large players to global trade governance, the EU must take the lead.
      </div>
    </div>
  </div>
</jest>
`;

exports[`Accordion renders correctly with extra attributes 1`] = `
<jest>
  <div
    class="ecl-accordion"
    data-ecl-accordion=""
    data-ecl-auto-init="Accordion"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <div
      class="ecl-accordion__item"
    >
      <h3
        class="ecl-accordion__title"
        id="accordion-example-title"
      >
        <button
          aria-controls="accordion-example-content"
          class="ecl-accordion__toggle"
          data-ecl-accordion-toggle=""
          type="button"
        >
          <span
            class="ecl-accordion__toggle-flex"
          >
            <span
              class="ecl-accordion__toggle-title"
            >
              Delivery of last pending proposals, a common Destiny of unity, the hour of European Democracy
            </span>
            <span
              class="ecl-accordion__toggle-indicator"
            >
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-accordion__toggle-icon"
                data-ecl-accordion-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#plus"
                />
              </svg>
            </span>
          </span>
        </button>
      </h3>
      <div
        aria-labelledby="accordion-example-title"
        class="ecl-accordion__content"
        hidden=""
        id="accordion-example-content"
        role="region"
      >
        The College of Commissioners held today the first weekly meeting of 2019 which was devoted to discussing the challenges of this new year. Commissioners used the opportunity to take stock and discuss the year ahead, including the European elections in May and other important milestones ahead.
      </div>
    </div>
    <div
      class="ecl-accordion__item"
    >
      <h3
        class="ecl-accordion__title"
        id="accordion-example2-title"
      >
        <button
          aria-controls="accordion-example2-content"
          class="ecl-accordion__toggle"
          data-ecl-accordion-toggle=""
          type="button"
        >
          <span
            class="ecl-accordion__toggle-flex"
          >
            <span
              class="ecl-accordion__toggle-title"
            >
              Spring 2019 Economic Forecast: Growth continues at a more moderate pace
            </span>
            <span
              class="ecl-accordion__toggle-indicator"
            >
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-accordion__toggle-icon"
                data-ecl-accordion-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#plus"
                />
              </svg>
            </span>
          </span>
        </button>
      </h3>
      <div
        aria-labelledby="accordion-example2-title"
        class="ecl-accordion__content"
        hidden=""
        id="accordion-example2-content"
        role="region"
      >
        The European economy is forecast to continue expanding for the seventh year in a row in 2019, with real GDP expected to grow in all EU Member States. As global uncertainties continue to weigh, domestic dynamics are set to support the European economy. Growth is expected to gather pace again next year.
      </div>
    </div>
    <div
      class="ecl-accordion__item"
    >
      <h3
        class="ecl-accordion__title"
        id="accordion-example3-title"
      >
        <button
          aria-controls="accordion-example3-content"
          class="ecl-accordion__toggle"
          data-ecl-accordion-toggle=""
          type="button"
        >
          <span
            class="ecl-accordion__toggle-flex"
          >
            <span
              class="ecl-accordion__toggle-title"
            >
              Delivery of last pending proposals, a common Destiny of unity, the hour of European Democracy
            </span>
            <span
              class="ecl-accordion__toggle-indicator"
            >
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-accordion__toggle-icon"
                data-ecl-accordion-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#plus"
                />
              </svg>
            </span>
          </span>
        </button>
      </h3>
      <div
        aria-labelledby="accordion-example3-title"
        class="ecl-accordion__content"
        hidden=""
        id="accordion-example3-content"
        role="region"
      >
        In the modern global economy trade is essential for growth, jobs and competiveness, and the EU is committed to maintaining an open and rules-based trading system. With the rising threat of protectionism and weakened commitment of large players to global trade governance, the EU must take the lead.
      </div>
    </div>
  </div>
</jest>
`;

exports[`Accordion renders correctly with extra class names 1`] = `
<jest>
  <div
    class="ecl-accordion custom-class custom-class--test"
    data-ecl-accordion=""
    data-ecl-auto-init="Accordion"
  >
    <div
      class="ecl-accordion__item"
    >
      <h3
        class="ecl-accordion__title"
        id="accordion-example-title"
      >
        <button
          aria-controls="accordion-example-content"
          class="ecl-accordion__toggle"
          data-ecl-accordion-toggle=""
          type="button"
        >
          <span
            class="ecl-accordion__toggle-flex"
          >
            <span
              class="ecl-accordion__toggle-title"
            >
              Delivery of last pending proposals, a common Destiny of unity, the hour of European Democracy
            </span>
            <span
              class="ecl-accordion__toggle-indicator"
            >
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-accordion__toggle-icon"
                data-ecl-accordion-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#plus"
                />
              </svg>
            </span>
          </span>
        </button>
      </h3>
      <div
        aria-labelledby="accordion-example-title"
        class="ecl-accordion__content"
        hidden=""
        id="accordion-example-content"
        role="region"
      >
        The College of Commissioners held today the first weekly meeting of 2019 which was devoted to discussing the challenges of this new year. Commissioners used the opportunity to take stock and discuss the year ahead, including the European elections in May and other important milestones ahead.
      </div>
    </div>
    <div
      class="ecl-accordion__item"
    >
      <h3
        class="ecl-accordion__title"
        id="accordion-example2-title"
      >
        <button
          aria-controls="accordion-example2-content"
          class="ecl-accordion__toggle"
          data-ecl-accordion-toggle=""
          type="button"
        >
          <span
            class="ecl-accordion__toggle-flex"
          >
            <span
              class="ecl-accordion__toggle-title"
            >
              Spring 2019 Economic Forecast: Growth continues at a more moderate pace
            </span>
            <span
              class="ecl-accordion__toggle-indicator"
            >
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-accordion__toggle-icon"
                data-ecl-accordion-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#plus"
                />
              </svg>
            </span>
          </span>
        </button>
      </h3>
      <div
        aria-labelledby="accordion-example2-title"
        class="ecl-accordion__content"
        hidden=""
        id="accordion-example2-content"
        role="region"
      >
        The European economy is forecast to continue expanding for the seventh year in a row in 2019, with real GDP expected to grow in all EU Member States. As global uncertainties continue to weigh, domestic dynamics are set to support the European economy. Growth is expected to gather pace again next year.
      </div>
    </div>
    <div
      class="ecl-accordion__item"
    >
      <h3
        class="ecl-accordion__title"
        id="accordion-example3-title"
      >
        <button
          aria-controls="accordion-example3-content"
          class="ecl-accordion__toggle"
          data-ecl-accordion-toggle=""
          type="button"
        >
          <span
            class="ecl-accordion__toggle-flex"
          >
            <span
              class="ecl-accordion__toggle-title"
            >
              Delivery of last pending proposals, a common Destiny of unity, the hour of European Democracy
            </span>
            <span
              class="ecl-accordion__toggle-indicator"
            >
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-accordion__toggle-icon"
                data-ecl-accordion-icon=""
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#plus"
                />
              </svg>
            </span>
          </span>
        </button>
      </h3>
      <div
        aria-labelledby="accordion-example3-title"
        class="ecl-accordion__content"
        hidden=""
        id="accordion-example3-content"
        role="region"
      >
        In the modern global economy trade is essential for growth, jobs and competiveness, and the EU is committed to maintaining an open and rules-based trading system. With the rising threat of protectionism and weakened commitment of large players to global trade governance, the EU must take the lead.
      </div>
    </div>
  </div>
</jest>
`;
